src/item.cpp

```diff
@@ -5963,6 +5967,19 @@ bool item::is_money() const
     return ammo_types().count( ammotype( "money" ) );
 }
 
+bool item::is_software() const
+{
+    const itype_id &cont_value = type->default_container.value_or( "null" );
+    if( !cont_value.is_null() ) {
+        return item( cont_value ).is_software_storage();
+    }
```

Comment:
```suggestion
    if( const cata_optional<itype_id> &cont = type->default_container ) {
        return item( *cont ).is_software_storage();
    }
```

Just repeating this suggestion from my other comment which may now be hidden under an "Outdated" fold.

Reply:
```suggestion
    if( const cata::optional<itype_id> &cont = type->default_container ) {
        return item( *cont ).is_software_storage();
    }
```

Reply:
@jbytheway I rewrote your suggestion because of typo (`cata_optional` --> `cata::optional`). Actually, I don't really like such non-obvious code, but since you insist, let it be.